superset/assets/javascripts/profile/components/TableLoader.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { Table, Tr, Td } from 'reactable';
+import { Collapse } from 'react-bootstrap';
+import $ from 'jquery';
+
+const propTypes = {
+  dataEndpoint: React.PropTypes.string.isRequired,
+  loadingNode: React.PropTypes.node,
+  mutator: React.PropTypes.func,
+  columns: React.PropTypes.arrayOf(React.PropTypes.string),
+};
+const defaultProps = {
+  loadingNode: <img alt="loading" width="25" src="/static/assets/images/loading.gif" />,
+};
+
+class TableLoader extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoading: true,
+      data: [],
+    };
+  }
+  componentWillMount() {
+    $.get(this.props.dataEndpoint, (data) => {
+      let actualData = data;
+      if (this.props.mutator) {
+        actualData = this.props.mutator(data);
+      }
+      this.setState({ data: actualData, isLoading: false });
+    });
+  }
+  render() {
+    const tableProps = Object.assign({}, this.props);
+    let columns = this.props.columns;
+    if (!columns && this.state.data.length > 0) {
+      columns = Object.keys(this.state.data[0]).filter(col => col[0] !== '_');
+    }
+    delete tableProps.loadingNode;
+    delete tableProps.dataEndpoint;
+    delete tableProps.columns;
+    if (this.state.isLoading) {
+      return this.props.loadingNode;
```

Comment:
i'm surprised the linter doesn't complain about multiple return statements. would be nice to follow the convention of a single return statement for the render function.